inflate.c

```diff
@@ -1002,21 +1062,26 @@ int32_t Z_EXPORT PREFIX(inflate)(PREFIX3(stream) *strm, int32_t flush) {
      */
   inf_leave:
     RESTORE();
+    int windowup = 0;
     if (INFLATE_NEED_UPDATEWINDOW(strm) &&
             (state->wsize || (out != strm->avail_out && state->mode < BAD &&
-                 (state->mode < CHECK || flush != Z_FINISH)))) {
+                 (state->mode < CHECK)))) {
```

Comment:
Does it work when you add back `flush != Z_FINISH`? I wonder if removing that is really needed..

Reply:
Seems to.

Reply:
Adding it backs seems to have a very small, barely measurable performance improvement on PNG decoding at least.  Whatever that's worth.  Though, it does look like a hack that was added in place to accommodate not wanting to run updatewindow on the last iteration or something.